Tests/Update/TestUpdate.swift

```diff
@@ -14,18 +14,108 @@ import XCTest
 
 class UpdateTestCase: XCTestCase {
 
-    func testEmptyGraph() {
+    func testEmptyGraph() throws {
 
-        struct Package: Update.Package {
-            let path = ""
+        class Package: Update.Package {
+            let url = ""
             let version = Version(0,0,0)
+
+            func upgrade(to: Version) throws {}
+            func fetch() throws -> [Version] { return [] }
+            func deps() -> [(URL, Range<Version>)] { return [] }
+        }
+
+        class RootPackage: Update.Parent {
+            func deps() -> [(URL, Range<Version>)] { return [] }
+        }
+
+        let delta = try update(root: RootPackage(), find: { _ -> Package? in return nil })
+
+        XCTAssertTrue(delta.upgraded.isEmpty)
```

Comment:
`XCTAssert(delta.upgraded, [])` is nice to use as a convention for this (when inference works), because then if it ever fails in CI the log shows you more information about what went wrong.